addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -967,9 +967,18 @@ export class Rtc {
         );
         this.state.channel.rtcInvitingSession = undefined;
         await this.loadSfuClient();
+        if (!this.state.channel?.id) {
+            return;
+        }
         await this.call();
+        if (!this.state.channel?.id) {
+            return;
+        }
         this.soundEffectsService.play("channel-join");
         await this.resetAudioTrack({ force: true });
+        if (!this.state.channel?.id) {
+            return;
+        }
```

Comment:
Ok as a fix for 17.0.

In master we should not rely on this `.id` check from detecting record is deleted. Instead, `state.channel` should become a relational field, so that we just have to check truthy on relational value.